### ISSUE TITLE
Single quotes not changed to double quotes

### ISSUE THREAD

**Issue by glen-84** [prettier/plugin-php#1254]
**@prettier/plugin-php v0.11.2**
[Playground link](https://loilo.github.io/prettier-php-playground/#N4IgDgTgpgLjCWUIHkwIPYDsDOIBcok8mMA6vACYwAW+AHAAwA0IMAhgEblW14AsLAK7YoAFU648AMzYAbES2zEA5rKgBFQehhR8M+VBYwIbeLJUBhdAFtrbAAoAJe-hCYsulhxMBjKAGUYAE81VzBsCABaACYQFmgAR0F4aHsTZTs9OQUQYhEIGDS2DLYsgwBfFigKeBh0FDR4LElQNmwYMpFK3MwwQQ68EAAeAH4wajAAHUxpgBI2AAIAXgWAcjZVgG45jmWFyZAOA82QcqA)

**Input:**
```php
<?php

$a = 'a';
$b = "b";
```

**Output:**
```php
<?php

$a = 'a';
$b = "b";
```

**Expected:**
```php
<?php

$a = "a";
$b = "b";
```

[JavaScript example](https://prettier.io/playground/#N4Igxg9gdgLgprEAuc0DOMAEBDTBeTAcm0IG4AdKSKDTAI303JDudJABoQIAHGAS3TJQ2AE6iIAdwAKYhGmQhsANwj8AJpxajsYANZwYAZR67+UAObIYogK5wuACxgBbADYB1R-3hpTYOCN5H35lHwBPRTA0BS5zNDhRGGkdCxdsZAAzbDcErgArNAAPACEdfUMjbBc4ABlzOCycvJBCoqNzCzc4AEVbCHgm3IcQU1EE0UU6bDo4Ny0eUXMYDw0YR2QADgAGLkWIBI8dHkVFuAnlRq4AR374FN4FFGw0AFooODh1L61ROFv+H8Utg0hkkNlhlwEi5+NY7CM0J1un0Bo1wc0RjAZqt1OtkAAmLg2bD8NydADCEBc6UUUGgVxAtgSABUZk8IS1lPYAJJQb6wIxgJZ8ACCfKMMHC3SGCQAvrKgA)

Related: #633.

**Comment by czosel** [comment]
This works when you activate the `single-quote` option, see [this example](https://loilo.github.io/prettier-php-playground/#N4IgDgTgpgLjCWUIHkwIPYDsDOIBcok8mMA6vACYwAW+AHAAwA0IMAhgEblW14AsLAK7YoAFU648AMzYAbES2zEA5rKgBFQehhR8MCIKgt9beLJUBhdAFtrbAAoAJe-hCYsulhwhsAxlABlGABPNVcwbAgAWgAmEBZoAEdBeGh7H2U7fBl5IxBiEQgYdLZMtmy5EQBfFigKeBh0FDR4LElQNmwYCtya-MwwQW68EAAeAH4wajAAHUw5gBI2AAIAXmWAcjYNgG5FjjXlmZAOY52QKqA) in the playground.

**Comment by glen-84** [comment]
@czosel I expect double quotes when the `single-quote` option is set to `false` (default), which is how it works for JS/TS.

**Comment by czosel** [comment]
@glen-84 Sorry, I misread your issue. I'm not sure if that's something that we can / should do, because double quotes in PHP are more like template literals in JS.

cc @evilebottnawi @loilo

**Comment by glen-84** [comment]
Ah, I see the issue now:

If:

```php
echo 'Hello $world';
```
... is automatically changed to:

```php
echo "Hello $world";
```

... then the behaviour of the application changes.

You could do the inverse of the single-quote handling though, i.e. if the string is "simple", then it can be changed.

**Comment by czosel** [comment]
Exactly! Yep, that might work. For reference, [here](https://github.com/prettier/plugin-php/blob/dc70b0710d5538fcf7a367f671269244f39aa759/src/util.js#L561) is the relevant function.

**Comment by Ralle** [comment]
You could also look for symbols inside the string like `$`, `\t`, `\n` and determine whether it can safely be changed or should be left alone.

**Comment by czosel** [comment]
Yeah, we could consider making it "symmetric" like that. What do you think @loilo, @evilebottnawi ?

**Comment by loilo** [comment]
As far as I know (not having the source at hand though), there are no relevant performance implications.

I think converting to double quotes would be reasonable for the sake of consistency. It was one of the first things I wondered about with this plugin, but I thought you just couldn't bother to detect whether a string can safely be converted to double quotes.

**Comment by czosel** [comment]
Right - since we're doing this detection already for double -> single, we could probably use it similarly for the other direction as well.

**Comment by utkarshgupta137** [comment]
Any updates?

**Comment by czosel** [comment]
Hi @utkarshgupta137 

No one has worked on this yet, but I think the change should be relatively straight-forward. If you'd be interested in contributing I'm happy to help you get started; otherwise I can try to take a look myself, but can't give any guarantees about when I find the time.

**Comment by utkarshgupta137** [comment]
Sure, let's do this.

**Comment by czosel** [comment]
Cool! https://github.com/prettier/plugin-php/blob/master/CONTRIBUTING.md should explain how to get started - let me know how I can help in case you get stuck!

**Comment by czosel** [comment]
Fixed and release as `v0.16.0` :tada:
